executor/distsql.go

```diff
@@ -346,6 +348,11 @@ func (e *TableReaderExecutor) Next(goCtx goctx.Context) (Row, error) {
 			e.feedback.Invalidate()
 			return nil, errors.Trace(err)
 		}
+		//TODO: isn't better we let its caller handle this situation?
+		// If partial result return a nil row and a nil error, it means
+		// we already drained all result.
+		// To me, it is kind of wired using a loop to do the logic
```

Comment:
To subjective statement, note that TiDB is a cooperative project which is stared by many people, it's not a personal toy project.

Reply:
B.T.W, you can tiny refine the code to make it  more readable, as you like.

Reply:
It is a question. I did not intend to leave this comment and merge into master. So per your understanding, is it normal here?